pkg/controller/serviceexternalip/controller.go

```diff
@@ -275,6 +401,19 @@ func getServiceExternalIP(service *corev1.Service) string {
 	return service.Status.LoadBalancer.Ingress[0].IP
 }
 
+func getServiceExternalIPPool(service *corev1.Service) string {
+	return service.Annotations[antreaagenttypes.ServiceExternalIPPoolAnnotationKey]
+}
+
+func isServiceExternalIPSharable(service *corev1.Service) bool {
+	value, exists := service.Annotations[antreaagenttypes.ServiceAllowSharedIPAnnotationKey]
+	if !exists {
+		return false
+	}
+	sharable, _ := strconv.ParseBool(value)
```

Comment:
There are some [well known annotation](https://kubernetes.io/docs/reference/labels-annotations-taints/) in K8s using boolean values. So I think users already know how to annotate it correctly. There also seems no good way to perform validation for annotation. We can log an invalid annotation in log but it wouldn't provide more information than the log "The Service didn't allow shared IP but the requested IP had been allocated to other Service"

Reply:
Got it, yes I think there is no great way. In case of an invalid annotation (e.g., typo), sharing will be disabled for the Service and users can figure it out when the behavior is not what they expect.